diff --git a/dist/runner.mjs b/dist/runner.mjs
index f4c6c95e8e65391c140168e81ff553a199a395f2..277c76f906f726700686e350b26047e857e471e1 100644
--- a/dist/runner.mjs
+++ b/dist/runner.mjs
@@ -2,7 +2,7 @@ import { logDebug, setupCore, teardownCore, getGitDir, optimizeFunction, mongoMe
 import path from 'path';
 import { chai } from 'vitest';
 import { NodeBenchmarkRunner } from 'vitest/runners';
-import { getHooks, getBenchFn } from 'vitest/suite';
+import { getHooks, getBenchFn, getBenchOptions } from 'vitest/suite';
 
 function getSuiteHooks(suite, name) {
   return getHooks(suite)[name];
@@ -26,7 +26,10 @@ async function runBench(benchmark, currentSuiteName) {
   const fn = getBenchFn(benchmark);
   await callSuiteHook(benchmark.suite, benchmark, "beforeEach");
   try {
-    await optimizeFunction(fn);
+    const options = getBenchOptions(benchmark);
+    if (options.warmupIterations !== 0) {
+      await optimizeFunction(fn);
+    }
   } catch (e) {
     if (!(e instanceof chai.AssertionError)) {
       throw e;
